File: backend/ws/message.go
package ws

import "nhooyr.io/websocket"

type Message struct {
	UserId                string `json:"user_id"`
	RoomId                string `json:"room_id"`
	For                   string `json:"for"`
	websocket.MessageType `json:"message_type"`
	Body                  any `json:"body"`
}

const (
	MSG   = "message"
	JOIN  = "join"
	LEFT  = "left"
	ERROR = "error"
)

File: frontend/mobile/chatapp/lib/message.dart
class Message {
  final String userId;
  final String type;
  final int messageType;
  final String body;

  Message({
    required this.userId,
    required this.type,
    required this.messageType,
    required this.body,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      userId: json['user_id'],
      type: json['for'],
      messageType: json['message_type'],
      body: json['body'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'for': type,
      'message_type': messageType,
      'body': body,
    };
  }
}
